### ISSUE TITLE
Serialization to a Folder does not preserve the relationship GUID

### ISSUE THREAD

**Issue by tobimax** [TabularEditor/TabularEditor#834]
Hi Daniel,

I'm not sure if this is a recent change or a configuration option or it just used to use new GUID's, however if you save to a folder the current relationship GUID is not persisted. This is causing issues when comparing models between different environments as the relationships never have the same id's from one deployment to another.

Current behavior in 2.16.0

1. Save model to folder
2. Load model from folder
3. Deploy model to SSAS/Power BI
4. Relationship Names are now called "Relationship 1", "Relationship 2" etc. (I'm sure this used to create a new GUID)
5. If you cut and paste the relationships in TE they all get new GUID's
6. If you cut and paste relationships with GUID's TE still generates new GUID's.

Expected behavior 

1. Unless their is a good reason not to, could the original relationship GUID be persisted when serializing to and from a folder.

Many Thanks 

Toby.

**Comment by otykier** [comment]
Tabular Editor uses the ID of the Relationship as the name of the file when serializing to a folder.

I agree that your item 5 and 6 is a bug (cutting and pasting should retain the original relationship ID), and I will fix these ASAP.

Hwoever, I weren't able to reproduce any issues/sudden renaming of relationships by following your steps 1-4. Could you clarify at which point the relationship IDs are changed? Thanks!

**Comment by otykier** [comment]
Item 5 and 6 are fixed in [2.16.1](https://github.com/otykier/TabularEditor/releases/tag/2.16.1). Leaving this open to await your reply on my question above.

**Comment by tobimax** [comment]
Hi Daniel,

Apologies for the delay in getting back to you. Steps 1 - 4 only happens if the Option **Serialize relationships by their starting table** which them embeds the JSON inside the Table file which has no relationship ID. 

```json
{
    "name": "Quality RMA Line (Fact)",
    "annotations": [
        {
            "name": "PBI_NavigationStepName",
            "value": "Navigation"
        },
        {
            "name": "PBI_ResultType",
            "value": "Table"
        },
        {
            "name": "TabularEditor_Relationships",
            "value": [
                "[",
                "  {",
                "    \"fromTable\": \"Quality RMA Line (Fact)\",",
                "    \"fromColumn\": \"CompanyID\",",
                "    \"toTable\": \"Companies\",",
                "    \"toColumn\": \"CompanyID\"",
                "  },",
                "  {",
                "    \"fromTable\": \"Quality RMA Line (Fact)\",",
                "    \"fromColumn\": \"CompanyID_Parent\",",
                "    \"toTable\": \"Companies\",",
                "    \"toColumn\": \"CompanyID\",",
                "    \"isActive\": false",
                "  },",
                "  {",
                "    \"fromTable\": \"Quality RMA Line (Fact)\",",
                "    \"fromColumn\": \"CompanyID_Responsible\",",
                "    \"toTable\": \"Companies\",",
                "    \"toColumn\": \"CompanyID\",",
                "    \"isActive\": false",
                "  },",
                "]"
            ]
        }
    ]
}
```

When you load the file back from disk and inspect the relationships using copy/paste they look like this.

```json
{
    "InstanceID": "00ec1cf6-8edc-4fd9-84d4-07b7e15edab0",
    "singlecolumnrelationships": [
        {
            "name": "Relationship 7",
            "fromTable": "Quality RMA Line (Fact)",
            "fromColumn": "CompanyID",
            "toTable": "Companies",
            "toColumn": "CompanyID"
        },
        {
            "name": "Relationship 8",
            "fromTable": "Quality RMA Line (Fact)",
            "fromColumn": "CompanyID_Parent",
            "toTable": "Companies",
            "toColumn": "CompanyID",
            "isActive": false
        },
        {
            "name": "Relationship 9",
            "fromTable": "Quality RMA Line (Fact)",
            "fromColumn": "CompanyID_Responsible",
            "toTable": "Companies",
            "toColumn": "CompanyID",
            "isActive": false
        }
    ]
}
```

Regards

Toby.
